### PR TITLE
indexation directory elements

### PR DESCRIPTION
**Pull request by ghazwarhili**

### CODE REVIEW

**Comment by SlimaneAmar** [comment]
Deal with deletion and modification elements
Add testContainer

File: src/main/java/org/gridsuite/directory/server/DirectoryService.java [hunk 1]
```diff
@@ -109,6 +112,21 @@ public ElementAttributes createElement(ElementAttributes elementAttributes, UUID
         assertElementNotExist(parentDirectoryUuid, elementAttributes.getElementName(), elementAttributes.getType());
         assertAccessibleDirectory(parentDirectoryUuid, userId);
         ElementAttributes result = insertElement(elementAttributes, parentDirectoryUuid);
+        UUID elementUuid = result.getElementUuid();
+        LocalDateTime lastModificationDate = result.getLastModificationDate().toLocalDateTime();
+        AccessRightsAttributes accessRights = result.getAccessRights();
+        UUID parentElementUuid = getParentElement(parentDirectoryUuid).getElementUuid();
+        directoryElementInfosService.addDirectoryElementInfos(
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/directory/server/DirectoryService.java:119]: Create a method `toDeirectoryElementInfos` in thje class `ElementAttributes`

File: src/main/java/org/gridsuite/directory/server/elasticsearch/DirectoryElementInfosService.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+ */
+@Service
+public class DirectoryElementInfosService {
+
+    private final DirectoryElementInfosRepository directoryElementInfosRepository;
+
+    @Value("${spring.data.elasticsearch.partition-size:10000}")
+    private int partitionSize;
+
+    public DirectoryElementInfosService(DirectoryElementInfosRepository directoryElementInfosRepository) {
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/directory/server/elasticsearch/DirectoryElementInfosService.java:28]: Remove all unnecessary methods and call directly  `directoryElementInfosRepository`
No code here for tests

File: src/test/java/org/gridsuite/directory/server/elasticsearch/DirectoryElementInfosServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,113 @@
+/*
```
Comment (SlimaneAmar) [inline src/test/java/org/gridsuite/directory/server/elasticsearch/DirectoryElementInfosServiceTest.java:1]: DisableElasticsearch on others tests

File: src/main/java/org/gridsuite/directory/server/elasticsearch/DirectoryElementInfosService.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        this.directoryElementInfosRepository = directoryElementInfosRepository;
+    }
+
+    public DirectoryElementInfos addDirectoryElementInfos(@NonNull DirectoryElementInfos elementInfos) {
+        return directoryElementInfosRepository.save(elementInfos);
+    }
+
+    public void addAllElementsInfos(@NonNull List<DirectoryElementInfos> directoryElementInfos) {
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/directory/server/elasticsearch/DirectoryElementInfosService.java:36]: `addAll` ?

**Review by SlimaneAmar — APPROVED** [review]

**Review by souissimai — APPROVED** [review]

**Pull request merged**